### PR TITLE
Adding timeout to synchronous functions

### PR DESCRIPTION
**Pull request by valleedelisle**
It might be interesting to tweak the timeout parameter on a per-function
level. This adds it to all functions that have a synchronous option.

### CODE REVIEW

**Comment by mshriver** [comment]
I think its a fine parameter to add, but there are also synchronous method implementations in EntityMixin classes, please include those in the update for a consistent API for synchronous entity methods.

A direct unit test would be great for adding confidence to the change.

Moving to draft state, please move back to ready to review when unit tests are passing and all method signatures/implementations have been updated.

**Review by mshriver — APPROVED** [review]
Unit test looks like adequate coverage and the changes look consistent across entities and the mixin functions, thanks for this enhancement!  I'd definitely like to see at least one more review by a tier2/maintainer and clean CI run before merging.

File: tests/test_entity_mixins.py
Comment (swadeley) [inline tests/test_entity_mixins.py:231]:
```suggestion
    def test_poll_task_timeout(self):
```
Was the double underscore intentional?

Comment (valleedelisle) [reply tests/test_entity_mixins.py:231]: yes it is because we're testing the `_poll_task` function.

**Review by JacobCallahan — APPROVED** [review]
ACK

**Pull request merged**
